CorsixTH/Lua/app.lua

```diff
@@ -1531,15 +1537,48 @@ end
 
 --! Function to check the loaded game is compatible with the program
 --!param save_version (num)
+--!param gfx_set (string) What graphics set is used
 --!return true if compatible, otherwise false
-function App:checkCompatibility(save_version)
+function App:checkCompatibility(save_version, gfx_set)
   local app_version = self.savegame_version
-  if app_version >= save_version or self.config.debug then
+  local err
+
+  -- First check the graphics set matches with the game files
+  if (gfx_set == "demo" and not self.using_demo_files) then
+    err = _S.errors.compatibility_error.demo_in_full
+  elseif (gfx_set == "full" and self.using_demo_files) then
+    err = _S.errors.compatibility_error.full_in_demo
+
+  -- if that's all good, check the save and app version
+  elseif app_version >= save_version or self.config.debug then
```

Comment:
I don't love that a middle branch is special in this if condition. It's a nit through. Will merge anyway.

Reply:
You could have

```lua
if demo and full then
  err = "error"
else if full and demo then
  err = "error"
end

if not err then 
  if (appver >= savver or config.debug) then
    return true
  else
    err = "error"
  end
end
```

Though not hot on that either...